src/main/java/frc/robot/command/AutoAlignCommand.java

```diff
@@ -0,0 +1,217 @@
+import edu.wpi.first.wpilibj2.command.Command;
+import edu.wpi.first.wpilibj2.command.ParallelRaceGroup;
+import edu.wpi.first.wpilibj2.command.SequentialCommandGroup;
+import edu.wpi.first.wpilibj2.command.WaitUntilCommand;
+import frc.robot.Constants;
+import frc.robot.input.ControllerInput;
+import frc.robot.subsystem.DriveSubsystem;
+import frc.robot.subsystem.PositioningSubsystem;
+import frc.robot.utils.Vec2;
+import com.pathplanner.lib.PathPlanner;
+
+/*
+ Driver holds down one of three buttons
+ The robot finds the closest master node
+ The robot generates a rotational and translational absolute trajectory to closest master node
+ The robot generates a translation relavative trajectory to the sub node based button held by the driver
+ The command ends when the final position is reached.
+ */
+public class AutoAlignCommand extends Command {
+  private final DriveSubsystem drive;
+  private final PositioningSubsystem positioning;
+  private final ControllerInput controller;
+
+  // Selected auto-align mode from the driver's button press
+  private AutoAlignMode selectedMode;
+
+  // Trajectories for the two phases
+  private Trajectory absoluteTrajectory;
+  private Trajectory relativeTrajectory;
+
+  // A sequential command that will follow both trajectories
+  private Command trajectoryCommand;
+
+  // Example master node positions (12 master nodes for 6 reef sides, 2 per side).
+  // These should normally be read from an external configuration (e.g. Elastic Dashboard)
+  // Here we provide dummy sample values (in meters) in field coordinates.
+  private final Pose2d[] masterNodes = new Pose2d[] {
+      new Pose2d(2.0, 7.0, Constants.ZERO_ROTATION),
+      new Pose2d(4.0, 7.0, Constants.ZERO_ROTATION),
+      new Pose2d(6.0, 7.0, Constants.ZERO_ROTATION),
+      new Pose2d(2.0, 5.0, Constants.ZERO_ROTATION),
+      new Pose2d(4.0, 5.0, Constants.ZERO_ROTATION),
+      new Pose2d(6.0, 5.0, Constants.ZERO_ROTATION),
+      new Pose2d(2.0, 3.0, Constants.ZERO_ROTATION),
+      new Pose2d(4.0, 3.0, Constants.ZERO_ROTATION),
+      new Pose2d(6.0, 3.0, Constants.ZERO_ROTATION),
+      new Pose2d(2.0, 1.0, Constants.ZERO_ROTATION),
+      new Pose2d(4.0, 1.0, Constants.ZERO_ROTATION),
+      new Pose2d(6.0, 1.0, Constants.ZERO_ROTATION)
+  };
+
+  // Mapping from auto-align mode to a relative offset for the final sub node.
+  // These offsets are translational adjustments (in meters) relative to the master node.
+  private final Map<AutoAlignMode, Translation2d> subNodeOffsets = new HashMap<>() {{
+      put(AutoAlignMode.CORAL_LEFT, new Translation2d(-0.3, 0.0));
+      put(AutoAlignMode.CORAL_RIGHT, new Translation2d(0.3, 0.0));
+      put(AutoAlignMode.ALGAE_MIDDLE, new Translation2d(0.0, 0.3));
+  }};
+
+  // Trajectory configuration taken from settings.json values (adjust as necessary)
+  private final TrajectoryConfig trajConfig = new TrajectoryConfig(3.0, 3.0)
+          .setKinematics(drive.kinematics);
```

Comment:
`TrajectoryConfig` is a class from the built-in wpilib trajectory generator. We don't use that, we use the 3rd-party [PathPlanner](https://pathplanner.dev/home.html) library.